Account/FxAPushMessageHandler.swift

```diff
@@ -108,57 +113,71 @@ extension FxAPushMessageHandler {
 
 /// An extension to handle each of the messages.
 extension FxAPushMessageHandler {
-    func handleDeviceConnected(_ data: JSON?) -> Success {
+    func handleDeviceConnected(_ data: JSON?) -> PushMessageResult {
         guard let deviceName = data?["deviceName"].string else {
             return messageIncomplete(.deviceConnected)
         }
-        return unimplemented(.deviceConnected, with: deviceName)
+        let message = PushMessage.deviceConnected(deviceName)
+        return deferMaybe(message)
     }
 }
 
 extension FxAPushMessageHandler {
-    func handleDeviceDisconnected(_ data: JSON?) -> Success {
+    func handleDeviceDisconnected(_ data: JSON?) -> PushMessageResult {
         guard let deviceID = data?["id"].string else {
             return messageIncomplete(.deviceDisconnected)
         }
-        return unimplemented(.deviceDisconnected, with: deviceID)
+
+        if deviceID == getOurClientId() {
+            // We can't disconnect the device from the account until we have
+            // access to the application, so we'll handle this properly in the AppDelegate,
+            // by calling the FxALoginHelper.applicationDidDisonnect(application).
+            profile.prefs.setBool(true, forKey: PendingAccountDisconnectedKey)
```

Comment:
Hmm. I separated disconnection out from this PR: this is a likely artifact of these being together at one point.